python-sdk/tests/test_storage.py

```diff
@@ -0,0 +1,168 @@
+import os
+import shutil
+from pathlib import Path
+from typing import Dict, Tuple
+
+import pytest
+
+from tests.fixtures import *
+
+from pachyderm_sdk.api import storage
+from .utils import count
+
+
+@pytest.fixture
+def fileset(client: TestClient) -> Tuple[str, Dict[str, bytes]]:
+    """Creates a fileset.
+
+    Returns
+    -------
+        (fileset_id, Dict[path, data])
+    """
+    files = {f"/file{i}": os.urandom(1024) for i in range(10)}
+    request_iterator = (
+        storage.CreateFilesetRequest(append_file=storage.AppendFile(path=file, data=data))
+        for file, data in files.items()
+    )
+    fileset = client.storage.create_fileset(request_iterator).fileset_id
+    return fileset, files
+
+
+class TestStorage:
+    """
+    Note:
+      As our tests and local development uses in-cluster object storage
+      (usually Minio), we need to patch the host of the presigned URL with
+      the correct, external host.
+    """
+
+    @staticmethod
+    def test_fetch_chunks(
+        client: TestClient,
+        fileset: Tuple[str, Dict[str, bytes]],
+        tmp_path: Path,
+    ):
+        """Test fetching chunks are stored in a local cache."""
+        # Arrange
+        fileset_id, _ = fileset
+
+        # Act
+        # This is hardcoded to replace the host of the presigned URL with
+        # localhost:9000. This test needs to be run simultaneously with:
+        #   - kubectl port-forward service/minio 9000:9000
+        client.storage.fetch_chunks(
+            fileset_id,
+            path="/",
+            cache_location=tmp_path,
+            http_host_replacement="localhost:9000",
+        )
+
+        # Assert
+        assert count(tmp_path.iterdir()) > 0
```

Comment:
Is it worth checking that the chunks follow the expected `f"{algorith.name.lower()}_{content_hash.hash.hex()}"` naming convention?

Reply:
Oh yeah you're totally right. I wrote this test before I went back and added the algorithm prefix to the chunk files.